### ISSUE TITLE
There is a bug in beforeStatementContinuationChars option of semi rule?

### ISSUE THREAD

**Issue by b6pzeusbc54tvhw5jgpyw8pwz2x6gs** [eslint/eslint#11053]
**Tell us about your environment**

* **ESLint Version:**
5.8.0

* **Node Version:**
8.12.0

* **npm Version:**
I did use yarn: yarn version: 1.10.1

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  extends: "standard",
  rules: {
    "space-before-function-paren": 'off',
    "keyword-spacing": 'off',
    "no-multiple-empty-lines": 'warn',
    "space-in-parens": 'off',
    "no-multiple-empty-lines": ['warn', { "max": 2, "maxEOF": 1 }],
    "comma-spacing": 'off',
    "semi": ['error', 'never', { beforeStatementContinuationChars:'always'}],
    "func-call-spacing": 'off',
    "no-unexpected-multiline": 'off',
    "no-unused-vars": 'warn',
    "no-sequences": 'off',
    "semi-style": ["error", "first"],
    indent: 'off',
  },
};
```

my devDependencies in package.json
```json
  "devDependencies": {
    "babel-plugin-styled-components": "^1.8.0",
    "eslint": "^5.8.0",
    "eslint-config-standard": "^12.0.0",
    "eslint-plugin-import": "^2.14.0",
    "eslint-plugin-node": "^8.0.0",
    "eslint-plugin-promise": "^4.0.1",
    "eslint-plugin-standard": "^4.0.0"
  }
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const something = () => {
  console.log('something')
}

something()
[1,2,3].forEach()
```

```bash
$ npx eslint something.js
```

**What did you expect to happen?**
`semi - beforeStatementContinuationChars: 'always'` option should detect error at `[1,2,3].forEach()`

**What actually happened? Please include the actual, raw output from ESLint.**
no error

**Are you willing to submit a pull request to fix this bug?**
No. Sorry, I'm eslint newbie.

**Comment by mysticatea** [comment]
Thank you for the report.

This is not a bug.
The `beforeStatementContinuationChars` option is about the statement that starts with continuation chars (e.g. `[`) but doesn't connect the previous statement.
For example:

```js
do {
    a = foo(a)
} while (a)
[1,2,3].forEach() //<- this statement doesn't connect the previous do-while loop.
```

I guess that you want [no-unexpected-multiline] rule to detect the statements that unexpectedly connected to the previous statement.

[no-unexpected-multiline]: https://eslint.org/docs/rules/no-unexpected-multiline
